nntrainer/compiler/tflite_interpreter.cpp

```diff
@@ -158,36 +183,74 @@ class TfOpIdxMap {
     buffer_map.addDataWhenNotFound(
       empty_buffer, {0, empty_buffer}); /// this represents empty buffer
 
-    auto &variable_map = getIndexMap<const float *, const Tensor *>();
-    auto update_variables = [&variable_map,
-                             &buffer_map](const TfOpNode::Variables &variables,
-                                          bool update_buffer) {
+    auto update_buffer_map = [&buffer_map](const TfOpNode::Variables &variables,
+                                           bool dynamic) {
       for (auto &variable : variables) {
         const float *buf = variable->getData();
-        variable_map.addDataWhenNotFound(buf, variable);
-        if (update_buffer) {
-          buffer_map.addDataWhenNotFound(buf, {variable->bytes(), buf});
-        }
+        assert(buf != nullptr);
+        auto byte_size = dynamic ? 0 : variable->bytes();
+        buffer_map.addDataWhenNotFound(buf, {byte_size, buf});
       }
     };
 
+    auto register_tensors =
+      [&tensors = this->tensors](const TfOpNode::Variables &variables) {
+        for (auto &variable : variables) {
+          auto tensor_it = std::find(tensors.begin(), tensors.end(), variable);
+          if (tensor_it == tensors.end()) {
+            tensors.push_back(variable);
+          }
+        }
+      };
+
     for (auto &op_node : nodes) {
+      if (op_node->isVirtualNode())
+        continue;
       update_opcode(op_node->getOpType());
-      update_variables(op_node->getInputs(), false);
-      update_variables(op_node->getWeights(), true);
-      update_variables(op_node->getOutputs(), false);
+
+      if (op_node->isInputNode()) {
+        /**
+         * Q) Why only register graph input tensor?
+         *
+         * A) the tflite needs only one tensor between nodes. Therefore,
+         *basically, no inputs are considered except graph input that doesn't
+         *have FROM node.
+         **/
+        register_tensors(op_node->getInputs());
```

Comment:
@zhoonit As you can see, `register_tensors` saves a tensor pointer to `tensors` vector. This `tensors` vector is used to export tflite tensor. AFAIK, in nntrainer, there are two tensors between `TfOpNode` that share its buffer. 

For example, say there are two nodes whose names are A and B. `B->getInputs()` would return a tensor named "BInput" and `A->getOutputs()` would return a tensor named "AOutput". "Binput" and "AOutput" are different tensor but they share the same buffer. In other word, there are two tensor between nodes. 

But, in tflite, as I've written in above comments, it needs only one tensor between nodes. So, I only add weights and outputs to `tensors` vector. And specifically, graph input nodes have only one tensor for each buffer because they are not the output of any other nodes.

Reply:
> "Binput" and "AOutput" are different tensor but they share the same buffer. In other word, there are two tensor between nodes.

"AOutput" and "BInput" should point to the same tensor in nntrainer as well, if it's not I guess there were some updates.